File: suc/kotlin/003.kt
package com.example.singlefileui

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ExperimentalLayoutApi
import androidx.compose.foundation.layout.FlowRow
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.WindowInsets
import androidx.compose.foundation.layout.aspectRatio
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.heightIn
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.AssistChip
import androidx.compose.material3.AssistChipDefaults
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.CenterAlignedTopAppBar
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.FilterChip
import androidx.compose.material3.FilterChipDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Shapes
import androidx.compose.material3.Surface
import androidx.compose.material3.Switch
import androidx.compose.material3.SwitchDefaults
import androidx.compose.material3.Text
import androidx.compose.material3.Typography
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.runtime.mutableIntStateOf
import androidx.compose.runtime.mutableStateListOf
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

private const val NAME = "006_DashboardAnalytics_en"
private const val UI_TYPE = "Dashboard Analytics"
private const val STYLE_THEME = "Corporate Dashboard"
private const val LANG = "en"
private const val BASELINE_SIZE = "1280x720"

object AppTokens {
    object Colors {
        val primary = Color(0xFF0EA5E9)
        val secondary = Color(0xFF22C55E)
        val tertiary = Color(0xFFF59E0B)
        val background = Color(0xFFF7FAFC)
        val surface = Color(0xFFFFFFFF)
        val surfaceVariant = Color(0xFFEFF3F7)
        val outline = Color(0xFFD4D9E1)
        val success = Color(0xFF16A34A)
        val warning = Color(0xFFF59E0B)
        val error = Color(0xFFEF4444)
        val onPrimary = Color(0xFFFFFFFF)
        val onSecondary = Color(0xFF0F172A)
        val onTertiary = Color(0xFF111827)
        val onBackground = Color(0xFF0B1220)
        val onSurface = Color(0xFF0B1220)
    }
    object TypographyTokens {
        val display = TextStyle(fontSize = 24.sp, fontWeight = FontWeight.SemiBold, lineHeight = 30.sp, letterSpacing = 0.sp)
        val headline = TextStyle(fontSize = 18.sp, fontWeight = FontWeight.SemiBold, lineHeight = 24.sp, letterSpacing = 0.sp)
        val title = TextStyle(fontSize = 14.sp, fontWeight = FontWeight.Medium, lineHeight = 18.sp, letterSpacing = 0.sp)
        val body = TextStyle(fontSize = 12.sp, fontWeight = FontWeight.Normal, lineHeight = 16.sp, letterSpacing = 0.sp)
        val label = TextStyle(fontSize = 10.sp, fontWeight = FontWeight.Medium, lineHeight = 14.sp, letterSpacing = 0.sp)
    }
    object Shapes {
        val small = RoundedCornerShape(6.dp)
        val medium = RoundedCornerShape(8.dp)
        val large = RoundedCornerShape(12.dp)
    }
    object Spacing {
        val xs = 2.dp
        val sm = 4.dp
        val md = 6.dp
        val lg = 8.dp
        val xl = 12.dp
        val xxl = 16.dp
        val xxxl = 24.dp
    }
    data class ShadowSpec(val elevation: Dp, val radius: Dp, val dy: Dp, val opacity: Float)
    object ElevationMapping {
        val level1 = ShadowSpec(1.dp, 2.dp, 0.5.dp, 0.12f)
        val level2 = ShadowSpec(2.dp, 4.dp, 1.dp, 0.14f)
        val level3 = ShadowSpec(3.dp, 6.dp, 1.5.dp, 0.16f)
        val level4 = ShadowSpec(4.dp, 8.dp, 2.dp, 0.18f)
        val level5 = ShadowSpec(5.dp, 10.dp, 2.5.dp, 0.2f)
    }
}

private val AppColorScheme = lightColorScheme(
    primary = AppTokens.Colors.primary,
    onPrimary = AppTokens.Colors.onPrimary,
    secondary = AppTokens.Colors.secondary,
    onSecondary = AppTokens.Colors.onSecondary,
    tertiary = AppTokens.Colors.tertiary,
    onTertiary = AppTokens.Colors.onTertiary,
    background = AppTokens.Colors.background,
    onBackground = AppTokens.Colors.onBackground,
    surface = AppTokens.Colors.surface,
    onSurface = AppTokens.Colors.onSurface,
    surfaceVariant = AppTokens.Colors.surfaceVariant,
    outline = AppTokens.Colors.outline,
    error = AppTokens.Colors.error
)

private val AppTypography = Typography(
    displayLarge = AppTokens.TypographyTokens.display,
    headlineMedium = AppTokens.TypographyTokens.headline,
    titleMedium = AppTokens.TypographyTokens.title,
    bodyMedium = AppTokens.TypographyTokens.body,
    labelMedium = AppTokens.TypographyTokens.label
)

@Composable
fun AppTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = AppColorScheme,
        typography = AppTypography,
        shapes = Shapes(
            extraSmall = AppTokens.Shapes.small,
            small = AppTokens.Shapes.small,
            medium = AppTokens.Shapes.medium,
            large = AppTokens.Shapes.large,
            extraLarge = AppTokens.Shapes.large
        ),
        content = content
    )
}

data class MetricCardUi(val id: Int, val title: String, val value: String, val delta: String, val color: Color)
data class ChartCardUi(val id: Int, val title: String, val color: Color)

@OptIn(ExperimentalMaterial3Api::class, ExperimentalLayoutApi::class)
@Composable
fun RootScreen() {
    val ranges = listOf("Today", "7 Days", "30 Days", "90 Days", "YTD", "Custom")
    val selectedRange = remember { mutableStateOf("7 Days") }
    val flags = remember { mutableStateListOf<String>() }
    val darkSwitch = remember { mutableStateOf(false) }
    val kpis = remember {
        listOf(
            MetricCardUi(1, "Revenue", "$24,560", "+8.2%", AppTokens.Colors.secondary),
            MetricCardUi(2, "Orders", "1,284", "+3.1%", AppTokens.Colors.primary),
            MetricCardUi(3, "Refunds", "42", "-1.4%", AppTokens.Colors.error),
            MetricCardUi(4, "Active Users", "8,921", "+5.6%", AppTokens.Colors.tertiary)
        )
    }
    val charts = remember {
        listOf(
            ChartCardUi(1, "Sales Trend", Color(0xFFE0F2FE)),
            ChartCardUi(2, "Conversion Funnel", Color(0xFFEFF6FF)),
            ChartCardUi(3, "Geo Distribution", Color(0xFFECFDF5)),
            ChartCardUi(4, "Device Breakdown", Color(0xFFFFFBEB))
        )
    }
    val notif = remember { mutableIntStateOf(2) }
    Scaffold(
        contentWindowInsets = WindowInsets(0),
        topBar = {
            CenterAlignedTopAppBar(
                title = {
                    Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm)) {
                        Box(modifier = Modifier.size(20.dp).background(AppTokens.Colors.primary, AppTokens.Shapes.small))
                        Text(text = "Analytics", style = MaterialTheme.typography.displayLarge, color = MaterialTheme.colorScheme.onSurface)
                    }
                }
            )
        },
        bottomBar = {
            Surface(color = MaterialTheme.colorScheme.surface, tonalElevation = AppTokens.ElevationMapping.level3.elevation, shadowElevation = AppTokens.ElevationMapping.level3.elevation) {
                Row(modifier = Modifier.fillMaxWidth().padding(horizontal = AppTokens.Spacing.lg, vertical = AppTokens.Spacing.md), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
                    Column(verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.xs), horizontalAlignment = Alignment.Start) {
                        Text(text = "Notifications", style = MaterialTheme.typography.labelMedium, color = MaterialTheme.colorScheme.onSurface)
                        Text(text = "${notif.intValue} pending", style = MaterialTheme.typography.headlineMedium, color = MaterialTheme.colorScheme.onSurface)
                    }
                    Button(onClick = { notif.intValue = 0 }, colors = ButtonDefaults.buttonColors(containerColor = MaterialTheme.colorScheme.secondary, contentColor = MaterialTheme.colorScheme.onSecondary), shape = AppTokens.Shapes.large, modifier = Modifier.height(44.dp).width(140.dp)) {
                        Text(text = "Mark All Read", style = MaterialTheme.typography.titleMedium)
                    }
                }
            }
        },
        containerColor = MaterialTheme.colorScheme.background
    ) { padding ->
        Column(modifier = Modifier.fillMaxSize().padding(padding).padding(horizontal = AppTokens.Spacing.lg, vertical = AppTokens.Spacing.md), verticalArrangement = Arrangement.Top, horizontalAlignment = Alignment.Start) {
            FlowRow(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm), verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.xs)) {
                ranges.forEach { r ->
                    val active = r == selectedRange.value
                    FilterChip(selected = active, onClick = { selectedRange.value = r }, label = { Text(text = r, style = if (active) MaterialTheme.typography.titleMedium else MaterialTheme.typography.bodyMedium, color = if (active) MaterialTheme.colorScheme.onPrimary else MaterialTheme.colorScheme.onSurface) }, colors = FilterChipDefaults.filterChipColors(containerColor = if (active) MaterialTheme.colorScheme.primary else MaterialTheme.colorScheme.surface, selectedContainerColor = MaterialTheme.colorScheme.primary), shape = AppTokens.Shapes.medium, border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline))
                }
            }
            Spacer(modifier = Modifier.height(AppTokens.Spacing.md))
            FlowRow(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm), verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.xs)) {
                Text(text = "Compare", style = MaterialTheme.typography.labelMedium, color = MaterialTheme.colorScheme.onSurface)
                AssistChip(onClick = { if (flags.contains("Prev Period")) flags.remove("Prev Period") else flags.add("Prev Period") }, label = { Text(text = "Prev Period", style = MaterialTheme.typography.labelMedium, color = if (flags.contains("Prev Period")) MaterialTheme.colorScheme.onSecondary else MaterialTheme.colorScheme.onSurface) }, shape = AppTokens.Shapes.small, border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline), colors = AssistChipDefaults.assistChipColors(containerColor = if (flags.contains("Prev Period")) MaterialTheme.colorScheme.secondary else MaterialTheme.colorScheme.surface))
                AssistChip(onClick = { if (flags.contains("Goal")) flags.remove("Goal") else flags.add("Goal") }, label = { Text(text = "Goal", style = MaterialTheme.typography.labelMedium, color = if (flags.contains("Goal")) MaterialTheme.colorScheme.onSecondary else MaterialTheme.colorScheme.onSurface) }, shape = AppTokens.Shapes.small, border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline), colors = AssistChipDefaults.assistChipColors(containerColor = if (flags.contains("Goal")) MaterialTheme.colorScheme.secondary else MaterialTheme.colorScheme.surface))
                Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.xs)) {
                    Text(text = "Dark", style = MaterialTheme.typography.labelMedium, color = MaterialTheme.colorScheme.onSurface)
                    Switch(checked = darkSwitch.value, onCheckedChange = { darkSwitch.value = it }, colors = SwitchDefaults.colors(checkedThumbColor = MaterialTheme.colorScheme.onPrimary, checkedTrackColor = MaterialTheme.colorScheme.primary))
                }
            }
            Spacer(modifier = Modifier.height(AppTokens.Spacing.lg))
            LazyVerticalGrid(columns = GridCells.Fixed(2), verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.lg), horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.lg), contentPadding = PaddingValues(bottom = AppTokens.Spacing.xxxl)) {
                items(kpis) { m ->
                    Card(shape = AppTokens.Shapes.large, colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface), elevation = CardDefaults.cardElevation(defaultElevation = AppTokens.ElevationMapping.level2.elevation), border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline), modifier = Modifier.fillMaxWidth().heightIn(min = 120.dp)) {
                        Column(modifier = Modifier.fillMaxWidth().padding(AppTokens.Spacing.lg), verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm), horizontalAlignment = Alignment.Start) {
                            Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
                                Text(text = m.title, style = MaterialTheme.typography.titleMedium, color = MaterialTheme.colorScheme.onSurface)
                                Box(modifier = Modifier.size(16.dp).background(m.color, AppTokens.Shapes.small))
                            }
                            Text(text = m.value, style = MaterialTheme.typography.headlineMedium, color = MaterialTheme.colorScheme.onSurface)
                            Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.xs)) {
                                Box(modifier = Modifier.size(6.dp).background(if (m.delta.startsWith("+")) AppTokens.Colors.success else AppTokens.Colors.error, AppTokens.Shapes.small))
                                Text(text = m.delta, style = MaterialTheme.typography.labelMedium, color = if (m.delta.startsWith("+")) AppTokens.Colors.success else AppTokens.Colors.error)
                            }
                        }
                    }
                }
                items(charts) { c ->
                    Card(shape = AppTokens.Shapes.large, colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface), elevation = CardDefaults.cardElevation(defaultElevation = AppTokens.ElevationMapping.level2.elevation), border = BorderStroke(1.dp, MaterialTheme.colorScheme.outline), modifier = Modifier.fillMaxWidth().heightIn(min = 180.dp)) {
                        Column(modifier = Modifier.fillMaxWidth().padding(AppTokens.Spacing.lg), verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.md), horizontalAlignment = Alignment.Start) {
                            Text(text = c.title, style = MaterialTheme.typography.titleMedium, color = MaterialTheme.colorScheme.onSurface)
                            Box(modifier = Modifier.fillMaxWidth().aspectRatio(1.6f).background(c.color, AppTokens.Shapes.medium), contentAlignment = Alignment.Center) {
                                Row(horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm), verticalAlignment = Alignment.CenterVertically) {
                                    Box(modifier = Modifier.size(6.dp, 40.dp).background(AppTokens.Colors.primary, AppTokens.Shapes.small))
                                    Box(modifier = Modifier.size(6.dp, 30.dp).background(AppTokens.Colors.secondary, AppTokens.Shapes.small))
                                    Box(modifier = Modifier.size(6.dp, 48.dp).background(AppTokens.Colors.tertiary, AppTokens.Shapes.small))
                                    Box(modifier = Modifier.size(6.dp, 22.dp).background(AppTokens.Colors.outline, AppTokens.Shapes.small))
                                    Box(modifier = Modifier.size(6.dp, 32.dp).background(AppTokens.Colors.primary, AppTokens.Shapes.small))
                                    Box(modifier = Modifier.size(6.dp, 42.dp).background(AppTokens.Colors.secondary, AppTokens.Shapes.small))
                                }
                            }
                            Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.sm), verticalAlignment = Alignment.CenterVertically) {
                                Box(modifier = Modifier.size(8.dp).background(AppTokens.Colors.primary, AppTokens.Shapes.small))
                                Text(text = "Series A", style = MaterialTheme.typography.labelMedium, color = MaterialTheme.colorScheme.onSurface)
                                Box(modifier = Modifier.size(8.dp).background(AppTokens.Colors.secondary, AppTokens.Shapes.small))
                                Text(text = "Series B", style = MaterialTheme.typography.labelMedium, color = MaterialTheme.colorScheme.onSurface)
                            }
                        }
                    }
                }
            }
        }
    }
}

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        WindowCompat.setDecorFitsSystemWindows(window, false)
        val controller = WindowInsetsControllerCompat(window, window.decorView)
        controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        controller.hide(WindowInsetsCompat.Type.navigationBars())
        setContent {
            AppTheme {
                Surface(color = MaterialTheme.colorScheme.background) {
                    RootScreen()
                }
            }
        }
    }
    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (hasFocus) {
            val controller = WindowInsetsControllerCompat(window, window.decorView)
            controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
            controller.hide(WindowInsetsCompat.Type.navigationBars())
        }
    }
}

@Preview(showBackground = true, backgroundColor = 0xFFFFFFFF, widthDp = 360, heightDp = 640)
@Composable
fun PreviewRoot() {
    AppTheme {
        RootScreen()
    }
}

File: suc/swift/003.swift
//
//  ContentView.swift
//  1
//
//  Created by hang on 2025/10/12.
//

import SwiftUI

// MARK: - Design Tokens - Exact values from Kotlin
struct AppTokens {
    struct Colors {
        static let primary = Color(hex: 0xFF0EA5E9)
        static let secondary = Color(hex: 0xFF22C55E)
        static let tertiary = Color(hex: 0xFFF59E0B)
        static let background = Color(hex: 0xFFF7FAFC)
        static let surface = Color(hex: 0xFFFFFFFF)
        static let surfaceVariant = Color(hex: 0xFFEFF3F7)
        static let outline = Color(hex: 0xFFD4D9E1)
        static let success = Color(hex: 0xFF16A34A)
        static let warning = Color(hex: 0xFFF59E0B)
        static let error = Color(hex: 0xFFEF4444)
        static let onPrimary = Color(hex: 0xFFFFFFFF)
        static let onSecondary = Color(hex: 0xFF0F172A)
        static let onTertiary = Color(hex: 0xFF111827)
        static let onBackground = Color(hex: 0xFF0B1220)
        static let onSurface = Color(hex: 0xFF0B1220)
    }
    
    struct TypographyTokens {
        static let display = Font.system(size: 24, weight: .semibold)
        static let headline = Font.system(size: 18, weight: .semibold)
        static let title = Font.system(size: 14, weight: .medium)
        static let body = Font.system(size: 12, weight: .regular)
        static let label = Font.system(size: 10, weight: .medium)
    }
    
    struct Spacing {
        static let xs: CGFloat = 2
        static let sm: CGFloat = 4
        static let md: CGFloat = 6
        static let lg: CGFloat = 8
        static let xl: CGFloat = 12
        static let xxl: CGFloat = 16
        static let xxxl: CGFloat = 24
    }
    
    struct Shapes {
        static let small = RoundedRectangle(cornerRadius: 6)
        static let medium = RoundedRectangle(cornerRadius: 8)
        static let large = RoundedRectangle(cornerRadius: 12)
    }
}

extension Color {
    init(hex: UInt) {
        let red = Double((hex >> 16) & 0xFF) / 255.0
        let green = Double((hex >> 8) & 0xFF) / 255.0
        let blue = Double(hex & 0xFF) / 255.0
        self.init(red: red, green: green, blue: blue)
    }
}

// MARK: - Data Models
struct MetricCardUi: Identifiable {
    let id: Int
    let title: String
    let value: String
    let delta: String
    let color: Color
}

struct ChartCardUi: Identifiable {
    let id: Int
    let title: String
    let color: Color
}

// MARK: - Main Content View
struct ContentView: View {
    @State private var selectedRange = "7 Days"
    @State private var selectedFlags: Set<String> = []
    @State private var darkMode = false
    @State private var notificationCount = 2
    
    private let ranges = ["Today", "7 Days", "30 Days", "90 Days"]
    private let ranges2 = ["YTD", "Custom"]
    private let flags = ["Prev Period", "Goal"]
    
    private let kpis: [MetricCardUi] = [
        MetricCardUi(id: 1, title: "Revenue", value: "$24,560", delta: "+8.2%", color: AppTokens.Colors.secondary),
        MetricCardUi(id: 2, title: "Orders", value: "1,284", delta: "+3.1%", color: AppTokens.Colors.primary),
        MetricCardUi(id: 3, title: "Refunds", value: "42", delta: "-1.4%", color: AppTokens.Colors.error),
        MetricCardUi(id: 4, title: "Active Users", value: "8,921", delta: "+5.6%", color: AppTokens.Colors.tertiary)
    ]
    
    private let charts: [ChartCardUi] = [
        ChartCardUi(id: 1, title: "Sales Trend", color: Color(hex: 0xFFE0F2FE)),
        ChartCardUi(id: 2, title: "Conversion Funnel", color: Color(hex: 0xFFEFF6FF))
    ]
    
    var body: some View {
        NavigationView {
            VStack(spacing: 0) {
                // Header - Exact replica of CenterAlignedTopAppBar
                HStack {
                    Spacer()
                    HStack(spacing: AppTokens.Spacing.sm) {
                        RoundedRectangle(cornerRadius: 4)
                            .fill(AppTokens.Colors.primary)
                            .frame(width: 20, height: 20)
                        
                        Text("Analytics")
                            .font(AppTokens.TypographyTokens.display)
                            .foregroundColor(AppTokens.Colors.onSurface)
                    }
                    Spacer()
                }
                .padding(.top, 48)
                .padding(.bottom, 12)
                .background(AppTokens.Colors.background)
                Divider()
                // Content
                ScrollView {
                    VStack(alignment: .leading, spacing: AppTokens.Spacing.md) {
                        // Date Range Selector
                        ScrollView(.horizontal, showsIndicators: false) {
                            HStack(spacing: AppTokens.Spacing.sm) {
                                ForEach(ranges, id: \.self) { range in
                                    Button(action: { selectedRange = range }) {
                                        Text(range)
                                            .font(selectedRange == range ? AppTokens.TypographyTokens.title : AppTokens.TypographyTokens.body)
                                            .foregroundColor(selectedRange == range ? AppTokens.Colors.onPrimary : AppTokens.Colors.onSurface)
                                            .padding(.horizontal, 16)
                                            .padding(.vertical, 8)
                                    }
                                    .background(
                                        AppTokens.Shapes.medium
                                            .fill(selectedRange == range ? AppTokens.Colors.primary : AppTokens.Colors.surface)
                                    )
                                    .overlay(
                                        AppTokens.Shapes.medium
                                            .stroke(AppTokens.Colors.outline, lineWidth: 1)
                                    )
                                }
                            }
                            .padding(.vertical, 4)
                            HStack(spacing: AppTokens.Spacing.sm) {
                                ForEach(ranges2, id: \.self) { range in
                                    Button(action: { selectedRange = range }) {
                                        Text(range)
                                            .font(selectedRange == range ? AppTokens.TypographyTokens.title : AppTokens.TypographyTokens.body)
                                            .foregroundColor(selectedRange == range ? AppTokens.Colors.onPrimary : AppTokens.Colors.onSurface)
                                            .padding(.horizontal, 16)
                                            .padding(.vertical, 10)
                                    }
                                    .background(
                                        AppTokens.Shapes.medium
                                            .fill(selectedRange == range ? AppTokens.Colors.primary : AppTokens.Colors.surface)
                                    )
                                    .overlay(
                                        AppTokens.Shapes.medium
                                            .stroke(AppTokens.Colors.outline, lineWidth: 1)
                                    )
                                }
                                Spacer()
                            }
                            .padding(.vertical, 8)
                        }
                        
                        
                        // Comparison Options
                        HStack(spacing: AppTokens.Spacing.sm) {
                            Text("Compare")
                                .font(AppTokens.TypographyTokens.label)
                                .foregroundColor(AppTokens.Colors.onSurface)
                            
                            ForEach(flags, id: \.self) { flag in
                                Button(action: {
                                    if selectedFlags.contains(flag) {
                                        selectedFlags.remove(flag)
                                    } else {
                                        selectedFlags.insert(flag)
                                    }
                                }) {
                                    Text(flag)
                                        .font(AppTokens.TypographyTokens.label)
                                        .foregroundColor(selectedFlags.contains(flag) ? AppTokens.Colors.onSecondary : AppTokens.Colors.onSurface)
                                        .lineLimit(1)
                                        .padding(.horizontal, 16)
                                        .padding(.vertical, 10)
                                }
                                .background(
                                    AppTokens.Shapes.small
                                        .fill(selectedFlags.contains(flag) ? AppTokens.Colors.secondary : AppTokens.Colors.surface)
                                )
                                .overlay(
                                    AppTokens.Shapes.small
                                        .stroke(AppTokens.Colors.outline, lineWidth: 1)
                                )
                            }
                            
                            // Dark Mode Toggle
                            HStack(spacing: AppTokens.Spacing.xs) {
                                Text("Dark")
                                    .font(AppTokens.TypographyTokens.label)
                                    .foregroundColor(AppTokens.Colors.onSurface)
                                Toggle(isOn: $darkMode){
                                    
                                }
                                .labelsHidden()
                                .toggleStyle(SwitchToggleStyle(tint: AppTokens.Colors.primary))
                                .scaleEffect(0.8)
                            }
                            .frame(maxWidth: .infinity, alignment: .leading)
                        }
                        .padding(.vertical, 6)
                        // Metrics Grid
                        LazyVGrid(
                            columns: [
                                GridItem(.flexible(), spacing: AppTokens.Spacing.lg),
                                GridItem(.flexible(), spacing: AppTokens.Spacing.lg)
                            ],
                            spacing: AppTokens.Spacing.lg
                        ) {
                            // First row: Revenue and Orders
                            MetricCardView(metric: kpis[0])
                            MetricCardView(metric: kpis[1])
                            
                            // Second row: Refunds and Active Users
                            MetricCardView(metric: kpis[2])
                            MetricCardView(metric: kpis[3])
                            
                            // Charts row
                            ChartCardView(chart: charts[0])
                            ChartCardView(chart: charts[1])
                        }
                        
                        Spacer().frame(height: 80)
                    }
                    .padding(.horizontal, AppTokens.Spacing.lg)
                    .padding(.vertical, AppTokens.Spacing.md)
                }
                .background(AppTokens.Colors.background)
            }
            
            // Bottom Notification Bar
            .overlay(alignment: .bottom) {
                VStack(spacing: 0) {
                    Rectangle()
                        .fill(AppTokens.Colors.outline.opacity(0.3))
                        .frame(height: 1)
                    
                    HStack {
                        VStack(alignment: .leading, spacing: AppTokens.Spacing.xs) {
                            Text("Notifications")
                                .font(AppTokens.TypographyTokens.label)
                                .foregroundColor(AppTokens.Colors.onSurface)
                            
                            Text("\(notificationCount) pending")
                                .font(AppTokens.TypographyTokens.headline)
                                .foregroundColor(AppTokens.Colors.onSurface)
                        }
                        
                        Spacer()
                        
                        Button(action: { notificationCount = 0 }) {
                            Text("Mark All Read")
                                .font(AppTokens.TypographyTokens.title)
                                .foregroundColor(AppTokens.Colors.onSecondary)
                                .frame(width: 140, height: 44)
                        }
                        .background(
                            AppTokens.Shapes.large
                                .fill(AppTokens.Colors.secondary)
                        )
                    }
                    .padding(.horizontal, AppTokens.Spacing.lg)
                    .padding(.vertical, AppTokens.Spacing.md)
                    .background(AppTokens.Colors.surface)
                    .shadow(color: .black.opacity(0.1), radius: 4, x: 0, y: -2)
                }
            }
            .navigationBarHidden(true) // 隐藏导航栏
        }
        .navigationViewStyle(StackNavigationViewStyle())
        .statusBar(hidden: true) // 隐藏状态栏
        .ignoresSafeArea() // 忽略安全区域
    }
}

// MARK: - Metric Card View
struct MetricCardView: View {
    let metric: MetricCardUi
    
    var body: some View {
        VStack(alignment: .leading, spacing: AppTokens.Spacing.sm) {
            HStack {
                Text(metric.title)
                    .font(AppTokens.TypographyTokens.title)
                    .foregroundColor(AppTokens.Colors.onSurface)
                
                Spacer()
                
                RoundedRectangle(cornerRadius: 4)
                    .fill(metric.color)
                    .frame(width: 16, height: 16)
            }
            
            Text(metric.value)
                .font(AppTokens.TypographyTokens.headline)
                .foregroundColor(AppTokens.Colors.onSurface)
            
            HStack(spacing: AppTokens.Spacing.xs) {
                RoundedRectangle(cornerRadius: 2)
                    .fill(metric.delta.hasPrefix("+") ? AppTokens.Colors.success : AppTokens.Colors.error)
                    .frame(width: 6, height: 6)
                
                Text(metric.delta)
                    .font(AppTokens.TypographyTokens.label)
                    .foregroundColor(metric.delta.hasPrefix("+") ? AppTokens.Colors.success : AppTokens.Colors.error)
            }
            Spacer()
        }
        .padding(AppTokens.Spacing.lg)
        .frame(height: 120)
        .frame(maxWidth: .infinity, alignment: .top)
        .background(
            AppTokens.Shapes.large
                .fill(AppTokens.Colors.surface)
                .shadow(color: .black.opacity(0.14), radius: 4, x: 0, y: 2)
        )
        .overlay(
            AppTokens.Shapes.large
                .stroke(AppTokens.Colors.outline, lineWidth: 1)
        )
        
    }
}

// MARK: - Chart Card View
struct ChartCardView: View {
    let chart: ChartCardUi
    
    var body: some View {
        VStack(alignment: .leading, spacing: AppTokens.Spacing.md) {
            Text(chart.title)
                .font(AppTokens.TypographyTokens.title)
                .foregroundColor(AppTokens.Colors.onSurface)
            
            ZStack {
                AppTokens.Shapes.medium
                    .fill(chart.color)
                
                // Bar chart representation
                HStack(spacing: AppTokens.Spacing.sm) {
                    RoundedRectangle(cornerRadius: 2)
                        .fill(AppTokens.Colors.primary)
                        .frame(width: 6, height: 40)
                    
                    RoundedRectangle(cornerRadius: 2)
                        .fill(AppTokens.Colors.secondary)
                        .frame(width: 6, height: 30)
                    
                    RoundedRectangle(cornerRadius: 2)
                        .fill(AppTokens.Colors.tertiary)
                        .frame(width: 6, height: 48)
                    
                    RoundedRectangle(cornerRadius: 2)
                        .fill(AppTokens.Colors.outline)
                        .frame(width: 6, height: 22)
                    
                    RoundedRectangle(cornerRadius: 2)
                        .fill(AppTokens.Colors.primary)
                        .frame(width: 6, height: 32)
                    
                    RoundedRectangle(cornerRadius: 2)
                        .fill(AppTokens.Colors.secondary)
                        .frame(width: 6, height: 42)
                }
            }
            .aspectRatio(1.6, contentMode: .fit)
            .frame(maxWidth: .infinity)
            
            // Chart legend
            HStack(spacing: AppTokens.Spacing.sm) {
                RoundedRectangle(cornerRadius: 2)
                    .fill(AppTokens.Colors.primary)
                    .frame(width: 8, height: 8)
                
                Text("Series A")
                    .font(AppTokens.TypographyTokens.label)
                    .foregroundColor(AppTokens.Colors.onSurface)
                
                RoundedRectangle(cornerRadius: 2)
                    .fill(AppTokens.Colors.secondary)
                    .frame(width: 8, height: 8)
                
                Text("Series B")
                    .font(AppTokens.TypographyTokens.label)
                    .foregroundColor(AppTokens.Colors.onSurface)
            }
        }
        .padding(AppTokens.Spacing.lg)
        .frame(height: 180)
        .frame(maxWidth: .infinity)
        .background(
            AppTokens.Shapes.large
                .fill(AppTokens.Colors.surface)
                .shadow(color: .black.opacity(0.14), radius: 4, x: 0, y: 2)
        )
        .overlay(
            AppTokens.Shapes.large
                .stroke(AppTokens.Colors.outline, lineWidth: 1)
        )
    }
}

// MARK: - Preview
struct ContentView_Previews: PreviewProvider {
    static var previews: some View {
        ContentView()
            .previewDevice("iPhone SE (3rd generation)")
    }
}

@main
struct DashboardApp: App {
    var body: some Scene {
        WindowGroup {
            ContentView() // Replace with your main view
        }
    }
}
